xkafka/config.go

```diff
@@ -0,0 +1,85 @@
+package xkafka
+
+import "time"
+
+// ConsumerConfig are configurations for creating a Kafka consumer.
+type ConsumerConfig struct {
+	Topics             []string      `yaml:"TOPICS" env:"TOPICS"`
```

Comment:
Thinking that config parsing should be done in all modules in xtools in the same way.

Currently, no other module offers config parsing from yaml/env, should we park this for now and take that up as a whole for the complete xtools repo?

Reply:
i am stuck between choosing an option based approach and config loader. Trying to figure out a trade off.

we can maybe start with options and later figure out config loading

Reply:
Having an Option API will allow a config struct to also be an option itself as long as it also implements the Option interface. I have seen this approach to be more flexible in such cases.